specification/under_development/profiles/Base_Profile.md

```diff
@@ -0,0 +1,484 @@
+; type definitions
+
+%Result = type opaque
+%Qubit = type opaque
+
+; global constants (labels for output recording)
+
+@0 = internal constant [3 x i8] c"r1\00"
+@1 = internal constant [3 x i8] c"r2\00"
+
+; entry point definition
+
+define i64 @Entry_Point_Name() #0 {
+entry:
+
+  ; calls to QIS functions
+  tail call void @__quantum__qis__h__body(%Qubit* null)
+  tail call void @__quantum__qis__cnot__body(%Qubit* null, %Qubit* inttoptr (i64 1 to %Qubit*))
+  tail call void @__quantum__qis__mz__body(%Qubit* null, writeonly %Result* null)
+  tail call void @__quantum__qis__mz__body(%Qubit* inttoptr (i64 1 to %Qubit*), writeonly %Result* inttoptr (i64 1 to %Result*))
+  br label %output
+
+output:                                   ; preds = %entry
+  ; calls to record the program output
+  tail call void @__quantum__rt__tuple_record_output(i64 2, i8* null)
+  tail call void @__quantum__rt__result_record_output(%Result* null, i8* getelementptr inbounds ([3 x i8], [3 x i8]* @0, i32 0, i32 0))
+  tail call void @__quantum__rt__result_record_output(%Result* inttoptr (i64 1 to %Result*), i8* getelementptr inbounds ([3 x i8], [3 x i8]* @1, i32 0, i32 0))
+
+  ret i64 0
+}
+
+; declarations of QIS functions
+
+declare void @__quantum__qis__h__body(%Qubit*)
+
+declare void @__quantum__qis__cnot__body(%Qubit*, %Qubit*)
+
+declare void @__quantum__qis__mz__body(%Qubit*, writeonly %Result*)
+
+; declarations of functions used for output recording
+
+declare void @__quantum__rt__tuple_record_output(i64, i8*)
+
+declare void @__quantum__rt__result_record_output(%Result*, i8*)
+
+; attributes
+
+attributes #0 = { "entry_point" "qir_profile"="base_profile" "output_labels"="schema_id" "required_qubits"="2" "required_results"="2" }
+
+; module flags
+
+!llvm.module.flags = !{!0, !1, !2, !3}
+
+!0 = !{i32 1, !"qir_major_version", i32 1}
+!1 = !{i32 7, !"qir_minor_version", i32 0}
+!2 = !{i32 1, !"dynamic_qubit_management", i1 false}
+!3 = !{i32 1, !"dynamic_result_management", i1 false}
+```
+
+The program entangles two qubits, measures them, and returns a tuple with the
+two measurement results.
+
+For the sake of clarity, the code above does not contain any [debug
+symbols](https://releases.llvm.org/13.0.0/docs/tutorial/MyFirstLanguageFrontend/LangImpl09.html?highlight=debug%20symbols).
+Debug symbols contain information that is used by a debugger to relate failures
+during execution back to the original source code. While we expect this to be
+primarily useful for execution on simulators, debug symbols are both easy to
+ignore and may be useful to generate helpful error messages for compilation
+failures that happen only late in the process. We hence see no reason to
+disallow them from occurring in the bitcode but will not detail their use any
+further. We defer to existing resources for more information about how to
+generate and use debug symbols.
+
+## Entry Point Definition
+
+The bitcode contains the definition of the LLVM function that should be invoked
```

Comment:
"The bitcode" => Here again, I wonder whether mentioning "bitcode" only is the right thing to do when the human-readable version is considered equivalent in LLVM logic.
"the LLVM function" => I have a doubt. Did we stick to having only one "internal" function? Do we mention that clearly somewhere here?

Reply:
Re bitcode: see my comment above - human readable and bitcode are not fully equivalent, and the requirement is for it to be bitcode.

Re one function: Yes, I believe we clearly state that the base profile will have only one function + declarations (but not definitions!) for the used QIS and RT functions.

Reply:
Saying that the bitcode and human readable form at NOT fully equivalent seems like poor practice to me.  Even if in current practical implementations this is a restriction due to test coverage not finding all the issues, why wouldn't you want to state that the two should map one-to-one as a stated goal?  In my opinion, this is a very important criterion for a robust system. But perhaps you just don't want to make it a requirement for the Base profile at this time?  Still ...

Reply:
@rtvuser1 I don't believe this is strictly within our power; this is up to LLVM - we are just "inheriting" the behavior here. I don't think we can or should try to limit what we use to the pieces that are strictly equivalent; I don't think we can predict evolution of LLVM in that regard.